mlflow/telemetry/track.py

```diff
@@ -24,7 +25,8 @@ def wrapper(*args, **kwargs):
         success = True
         start_time = time.time()
         try:
-            return func(*args, **kwargs)
+            with _avoid_telemetry_tracking():
```

Comment:
Maybe add a comment why we have this logic (e.g, "Avoid tracking nested api calles from databricks-agents or log_model")